tests/shared/nlu/training_data/test_training_data.py

```diff
@@ -633,3 +634,50 @@ def test_fingerprint_is_same_when_loading_data_again():
     td1 = training_data_from_paths(files, language="en")
     td2 = training_data_from_paths(files, language="en")
     assert td1.fingerprint() == td2.fingerprint()
+
+
+@pytest.mark.parametrize(
+    "intent_frequencies, num_chunks", [([100, 82, 63, 43], 8), ([15, 12, 10, 7], 4)]
+)
+def test_divide_training_data_chunks(intent_frequencies, num_chunks):
+
+    # Create the initial training data
+    all_messages = []
+    for index, intent_count in enumerate(intent_frequencies):
+        all_messages.extend(
+            [
+                Message(
+                    text=f"intent_{index * intent_count + ex_index}", intent=f"{index}"
+                )
+                for ex_index in range(intent_count)
+            ]
+        )
+    training_data = TrainingData(all_messages)
+    original_fingerprint = training_data.fingerprint()
+    chunks = training_data.divide_into_chunks(num_chunks=num_chunks)
+    new_fingerprint = training_data.fingerprint()
+
+    # Original training data shouldn't be modified
+    assert original_fingerprint == new_fingerprint
```

Comment:
Why? It should not be needed anymore after we created the chunks.

Reply:
This is to be sure that we haven't modified the original `TrainingData` instance in the process of dividing into chunks. I would say even if it's not needed, it should be still preserved. What do you think?

Reply:
ok, makes sense